crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -1064,16 +1064,22 @@ impl<P: RoomDataProvider> TimelineController<P> {
 
                     match decryptor.decrypt_event_impl(original_json).await {
                         Ok(event) => {
-                            trace!(
-                                "Successfully decrypted event that previously failed to decrypt"
-                            );
-
-                            // Notify observers that we managed to eventually decrypt an event.
-                            if let Some(hook) = unable_to_decrypt_hook {
-                                hook.on_late_decrypt(&remote_event.event_id, *utd_cause).await;
+                            if let matrix_sdk::deserialized_responses::TimelineEventKind::UnableToDecrypt { utd_info, ..} = event.kind {
```

Comment:
nit: use `use`, not the fully qualified name here

Reply:
> nit: use `use`, not the fully qualified name here

There is already an import with the same name `super::event_handler::TimelineEventKind` hence the need for fully qualified name.

Reply:
you can `use matrix_sdk::deserialized_responses::TimelineEventKind as SdkTimelineEventKind` or something, to give it a different name

Reply:
I like Rich's suggestion :+1:

Reply:
done